src/domain/Multigrid.cpp

```diff
@@ -607,3 +607,14 @@ bool Multigrid::is_obstacle_cell(const size_t level,
 std::vector<FieldType> Multigrid::get_used_fields() {
     return m_bdc_domain.get_used_fields();
 }
+
+bool Multigrid::is_blocked_by_obstacle(Coordinate<size_t> &start, Coordinate<size_t> &end) {
+    std::vector<Obstacle> obstacle_list = m_MG_obstacle_object_list[0];
+    for (size_t id = 0; id < m_number_of_obstacle_objects; id++) {
+        bool blocked = obstacle_list[id].intersection(start, end);
+        if (blocked) {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
You could rewrite this as:

```cpp
bool Multigrid:is_blocked_by_obstacle(const Coordinate<size_t>& from, const Coordinate<size_t>& to) const {
    const auto& obstacles = m_MG_obstacle_object_list[0];
    for(const auto& obstacle : obstacles) {
        if(obstacle.instersection(from, to)) {
            return true;
        }
    }
    return false
}
```

Note the original code was copying all obstacles in `std::vector<Obstacle> obstacle_list = m_MG_obstacle_object_list[0];` 
If I remember the design goal correctly this function is called frequently so allocation and copy should be avoided.